hwilib/devices/trezor.py

```diff
@@ -117,39 +117,108 @@ def func(*args, **kwargs):
             raise DeviceConnectionError('Device disconnected')
     return func
 
+
 def interactive_get_pin(self, code=None):
-    if code == PIN_CURRENT:
+    if code == messages.PinMatrixRequestType.Currrent:
         desc = "current PIN"
-    elif code == PIN_NEW:
+    elif code == messages.PinMatrixRequestType.NewFirst:
         desc = "new PIN"
-    elif code == PIN_CONFIRM:
+    elif code == messages.PinMatrixRequestType.NewSecond:
         desc = "new PIN again"
     else:
         desc = "PIN"
 
-    echo(PIN_MATRIX_DESCRIPTION)
+    print(PIN_MATRIX_DESCRIPTION, file=sys.stderr)
 
     while True:
-        pin = prompt("Please enter {}".format(desc), hide_input=True)
+        pin = getpass.getpass(f"Please entire {desc}:\n")
         if not pin.isdigit():
-            echo("Non-numerical PIN provided, please try again")
+            print("Non-numerical PIN provided, please try again", file=sys.stderr)
         else:
             return pin
 
+
+def mnemonic_words(expand=False, language="english"):
+    if expand:
+        wordlist = Mnemonic(language).wordlist
+    else:
+        wordlist = set()
+
+    def expand_word(word):
+        if not expand:
+            return word
+        if word in wordlist:
+            return word
+        matches = [w for w in wordlist if w.startswith(word)]
+        if len(matches) == 1:
+            return matches[0]
+        print("Choose one of: " + ", ".join(matches), file=sys.stderr)
+        raise KeyError(word)
+
+    def get_word(type):
+        assert type == messages.WordRequestType.Plain
+        while True:
+            try:
+                word = input("Enter one word of mnemonic:\n")
+                return expand_word(word)
+            except KeyError:
+                pass
+            except Exception:
+                raise Cancelled from None
+
+    return get_word
+
+
+class PassphraseUI:
+    def __init__(self, passphrase):
+        self.passphrase = passphrase
+        self.pinmatrix_shown = False
+        self.prompt_shown = False
+        self.always_prompt = False
+        self.return_passphrase = True
+
+    def button_request(self, code):
+        if not self.prompt_shown:
+            print("Please confirm action on your Trezor device", file=sys.stderr)
+        if not self.always_prompt:
+            self.prompt_shown = True
+
+    def get_pin(self, code=None):
+        raise NotImplementedError('get_pin is not needed')
+
+    def disallow_passphrase(self):
+        self.return_passphrase = False
+
+    def get_passphrase(self):
+        if self.return_passphrase:
+            return self.passphrase
+        raise ValueError('Passphrase from Host is not allowed for Trezor T')
+
+
+def get_path_transport(path: str):
```

Comment:
any reason why this exists, as opposed to `trezorlib.transport.get_transport` ?

Reply:
...right, because there's no pass-through in `get_transport` that would allow finding non-Trezor USB IDs. hence also the `usb_ids=None` thing

Reply:
I actually did this to avoid having to remove the `BridgeTransport` stuff from `all_transports`.

Reply:
that's easier to accomplish by setting BridgeTransport.ENABLED = False

Reply:
`trezorlib/transport/bridge.py` is specifically excluded from the local copy because it requires `requests`. Including it either requires us to add `requests` as a dependency, or modify the file to not do that. I do not want to do either of those things.

Reply:
so i'm wondering, do you intend to include `trezorlib` as a dependency once your changes are in, or are you still going to vendor a copy? because obviously including it as a dependency would pull in a lot of things like requests

Reply:
I'm still planning on vendoring a copy, however I would greatly prefer if all I had to do was copy over a specific set of files and not have to make changes to them.